wallet-new/src/Cardano/Wallet/Kernel/CoinSelection/FromGeneric.hs

```diff
@@ -178,6 +184,34 @@ feeOptions CoinSelectionOptions{..} = FeeOptions{
   Building transactions
 -------------------------------------------------------------------------------}
 
+-- | Our notion of @unsigned transaction@. Unfortunately we cannot reuse
+-- directly the 'Tx' from @Core@ as that discards the information about
+-- "ownership" of inputs, which is instead required when dealing with the
+-- Core Txp.Util API.
+data UnsignedTx = UnsignedTx {
+    utxOwnedInputs :: NonEmpty (Core.TxIn, Core.TxOutAux)
+  , utxOutputs     :: NonEmpty Core.TxOutAux
+  , utxChange      :: [Core.Coin]
+  , utxAttributes  :: Core.TxAttributes
+}
+
+-- | Creates a "standard" unsigned transaction.
+mkStdUnsignedTx :: Monad m
+             => (forall a. NonEmpty a -> m (NonEmpty a))
+             -- ^ Shuffle function
+             -> NonEmpty (Core.TxIn, Core.TxOutAux)
+             -- ^ Selected inputs
+             -> NonEmpty Core.TxOutAux
+             -- ^ Selected outputs
+             -> [Core.Coin]
+             -- ^ Change coins
+             -> m UnsignedTx
+mkStdUnsignedTx shuffle inps outs change = do
+    allOuts <- shuffle outs
```

Comment:
This shuffling here serves no purpose: the main goal of shuffling is to hide change addresses, but we cannot shuffle the change addresses yet here.